summary-bot の summarize job が GitHub Models の廃止で 410 になり CI が落ちる

## 事象

`.github/workflows/summary-bot.yml` の `summarize` job が失敗する。PR #151 の CI で観測した。

```
##[error]API error: Error: 410 GitHub Models is temporarily unavailable as part of a scheduled retirement brownout.
```

実行ログ: https://github.com/miyashita337/claude-context-manager/actions/runs/32095226817/job/95585194825

## 原因

`actions/ai-inference@v1` が `https://models.github.ai/inference`（GitHub Models）を叩いているが、GitHub Models が**廃止過程（scheduled retirement brownout）**に入り 410 を返している。brownout は本番廃止前の予行演習として一時的に停止させる運用なので、このまま放置すると恒久的に失敗するようになる。

同ログには別の警告もある。

```
##[warning]Node.js 20 is deprecated. The following actions target Node.js 20 but are being
forced to run on Node.js 24: actions/ai-inference@v1
```

## 影響

- PR の CI が `summarize` の 1 件だけ FAILURE になり、mergeStateStatus が UNSTABLE になる
- 他の check（Python/TypeScript のテスト・lint・Security Scan・Qodo・CodeRabbit・codecov）はすべて SUCCESS

## 対応案

| 案 | 内容 |
|---|---|
| A | `summary-bot.yml` を削除する。本リポには CodeRabbit と Qodo pr-agent（PR #151 で追加）があり、PR 要約は両方が投稿するため機能が重複している |
| B | LLM プロバイダを GitHub Models から別のものへ差し替える（例: Qodo と同じ Gemini API 無料枠） |
| C | 当該 job を `continue-on-error: true` にして CI を落とさないようにする（暫定） |

**A を推奨**する。Qodo pr-agent が PR 要約とレビューを投稿するようになったため、`summary-bot` の役割は実質的に代替されている（[agent-output-quality.md](https://github.com/miyashita337/agent-base) #4「使われない機能の継続保守の禁止」）。

## 発見経緯

Qodo pr-agent 導入 PR #151 の CI 確認中に発見した。本 Issue の事象は当該 PR の変更とは無関係で、既存 workflow の問題である。

--- a/.github/workflows/summary-bot.yml
+++ b/.github/workflows/summary-bot.yml
@@ -1,5 +1,16 @@
 name: Summary Bot
 
+# Issue / PR が開かれたら、内容を要約したポエムをコメントする。
+#
+# LLM は Google AI Studio の Gemini API（Flash 系は無料枠）を使う。
+# 以前は actions/ai-inference + GitHub Models だったが、GitHub Models が廃止過程
+# （scheduled retirement brownout）に入り 410 を返すようになり、本ジョブは 2026-08 以降
+# 何も投稿しない死んだ状態だった（Issue #152）。actions/ai-inference@v1 自体も
+# provider が Copilot CLI 専用へ移行しており、runner 上に認証済み Copilot CLI を
+# 要求するため代替にならない。よって API を直接叩く形に変更した。
+#
+# 必要な secret: GEMINI_API_KEY（Qodo pr-agent と共用）
+
 on:
   issues:
     types: [opened]
@@ -9,56 +20,85 @@ on:
 permissions:
   issues: write
   pull-requests: write
-  models: read
   contents: read
 
 jobs:
   summarize:
     runs-on: ubuntu-latest
+    # 要約は付加価値であり、失敗しても本体の CI を妨げない
     continue-on-error: true
     steps:
       - name: Generate poem summary
         id: ai
-        uses: actions/ai-inference@v1
-        with:
-          model: openai/gpt-4o-mini
-          max-tokens: 300
-          system-prompt: |
-            あなたは GitHub Issue/PR を読んで情報密度の高いポエム調要約を書くボットです。
+        env:
+          GEMINI_API_KEY: ${{ secrets.GEMINI_API_KEY }}
+          # ${{ }} を run: の中へ直接展開するとスクリプトインジェクションになるため env 経由で渡す
+          TITLE: ${{ github.event.issue.title || github.event.pull_request.title }}
+          BODY: ${{ github.event.issue.body || github.event.pull_request.body }}
+        run: |
+          set -euo pipefail
 
-            要件:
-            - 3〜6行で書く
-            - 何の問題/機能か、原因や対策など具体的な事実を必ず含める
-            - ポエム調・詩的なリズムで書く（比喩OKだが内容は具体的に）
-            - 読んだ人が「何の話か」が30秒でわかるレベルの情報量
-            - 日本語で書く
-            - 要約のみを出力し、前置きや説明は不要
+          if [ -z "${GEMINI_API_KEY:-}" ]; then
+            echo "::error::GEMINI_API_KEY secret is not set" >&2
+            exit 1
+          fi
 
-            良い例（日本語文字化けIssueの場合）:
-            ひらがなが+0x90オフセットでカタカナへ化ける怪、
-            IMEカーソルが日本語を飲み込む二つの罪。
-            accessibilitySupport設定で静めながら、
-            根本原因は闇の中、観察続ける。
-          prompt: |
-            タイトル: ${{ github.event.issue.title || github.event.pull_request.title }}
+          SYSTEM_PROMPT='あなたは GitHub Issue/PR を読んで情報密度の高いポエム調要約を書くボットです。
 
-            本文:
-            ${{ github.event.issue.body || github.event.pull_request.body }}
+          要件:
+          - 3〜6行で書く
+          - 何の問題/機能か、原因や対策など具体的な事実を必ず含める
+          - ポエム調・詩的なリズムで書く（比喩OKだが内容は具体的に）
+          - 読んだ人が「何の話か」が30秒でわかるレベルの情報量
+          - 日本語で書く
+          - 要約のみを出力し、前置きや説明は不要
+
+          良い例（日本語文字化けIssueの場合）:
+          ひらがなが+0x90オフセットでカタカナへ化ける怪、
+          IMEカーソルが日本語を飲み込む二つの罪。
+          accessibilitySupport設定で静めながら、
+          根本原因は闇の中、観察続ける。'
+
+          # 本文は長くなりうるので先頭 8000 文字に制限する（無料枠のトークン節約）
+          jq -n \
+            --arg system "$SYSTEM_PROMPT" \
+            --arg user "タイトル: ${TITLE}"$'\n\n'"本文:"$'\n'"$(printf '%s' "${BODY:-}" | head -c 8000)" \
+            '{
+               system_instruction: { parts: [ { text: $system } ] },
+               contents: [ { parts: [ { text: $user } ] } ],
+               generationConfig: { maxOutputTokens: 1024 }
+             }' > request.json
+
+          # gemini-3.7-flash は thinking にトークンを使うため maxOutputTokens に余裕を持たせる
+          # （2.5-flash は 300 だと finishReason=MAX_TOKENS で途中で切れることを実測済み）
+          HTTP_CODE=$(curl -sS -o response.json -w '%{http_code}' \
+            -X POST "https://generativelanguage.googleapis.com/v1beta/models/gemini-3.7-flash:generateContent?key=${GEMINI_API_KEY}" \
+            -H 'Content-Type: application/json' \
+            --data @request.json)
+
+          if [ "$HTTP_CODE" != "200" ]; then
+            echo "::error::Gemini API returned HTTP ${HTTP_CODE}" >&2
+            jq -r '.error.message // "no error message"' response.json >&2
+            exit 1
+          fi
+
+          FINISH=$(jq -r '.candidates[0].finishReason // "UNKNOWN"' response.json)
+          jq -r '.candidates[0].content.parts[0].text // ""' response.json > poem.txt
+
+          if [ "$FINISH" != "STOP" ] || [ ! -s poem.txt ]; then
+            echo "::error::unusable response (finishReason=${FINISH}, bytes=$(wc -c < poem.txt))" >&2
+            exit 1
+          fi
 
       - name: Post poem as comment
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NUMBER: ${{ github.event.issue.number || github.event.pull_request.number }}
-          RESPONSE: ${{ steps.ai.outputs.response }}
           REPO: ${{ github.repository }}
         run: |
           set -euo pipefail
-          if [ -z "${RESPONSE:-}" ]; then
-            echo "ai-inference returned empty response; skipping comment"
-            exit 0
-          fi
           {
             echo "> 🎭 *AIポエムサマリー*"
             echo ">"
-            printf '%s\n' "$RESPONSE" | sed 's/^/> /'
+            sed 's/^/> /' poem.txt
           } | gh issue comment "$NUMBER" --repo "$REPO" --body-file -

--- a/.github/workflows/summary-bot.yml
+++ b/.github/workflows/summary-bot.yml
@@ -25,8 +25,16 @@ permissions:
 jobs:
   summarize:
     runs-on: ubuntu-latest
+    # fork からの PR には repository secret が渡らないため、無駄に赤くせずスキップする。
+    # secrets context は jobs.<job_id>.if では参照できない仕様のため github context で判定する
+    # https://docs.github.com/en/actions/reference/workflows-and-actions/contexts
+    if: >-
+      github.event_name != 'pull_request' ||
+      github.event.pull_request.head.repo.full_name == github.repository
     # 要約は付加価値であり、失敗しても本体の CI を妨げない
     continue-on-error: true
+    # API / ネットワークが停滞しても runner を長時間占有しない
+    timeout-minutes: 5
     steps:
       - name: Generate poem summary
         id: ai
@@ -59,19 +67,25 @@ jobs:
           accessibilitySupport設定で静めながら、
           根本原因は闇の中、観察続ける。'
 
-          # 本文は長くなりうるので先頭 8000 文字に制限する（無料枠のトークン節約）
+          # 本文は長くなりうるので先頭 8000 文字に制限する（無料枠のトークン節約）。
+          # 切り詰めは jq の文字列スライスで行う。`head -c` はバイト単位で切るため、
+          # 日本語では文字数が合わないうえ UTF-8 のマルチバイト文字を途中で分断しうる
           jq -n \
             --arg system "$SYSTEM_PROMPT" \
-            --arg user "タイトル: ${TITLE}"$'\n\n'"本文:"$'\n'"$(printf '%s' "${BODY:-}" | head -c 8000)" \
+            --arg title "$TITLE" \
+            --arg body "${BODY:-}" \
             '{
                system_instruction: { parts: [ { text: $system } ] },
-               contents: [ { parts: [ { text: $user } ] } ],
+               contents: [ { parts: [ {
+                 text: ("タイトル: " + $title + "\n\n本文:\n" + $body[0:8000])
+               } ] } ],
                generationConfig: { maxOutputTokens: 1024 }
              }' > request.json
 
           # gemini-3.7-flash は thinking にトークンを使うため maxOutputTokens に余裕を持たせる
           # （2.5-flash は 300 だと finishReason=MAX_TOKENS で途中で切れることを実測済み）
           HTTP_CODE=$(curl -sS -o response.json -w '%{http_code}' \
+            --connect-timeout 10 --max-time 120 --retry 2 --retry-connrefused \
             -X POST "https://generativelanguage.googleapis.com/v1beta/models/gemini-3.7-flash:generateContent?key=${GEMINI_API_KEY}" \
             -H 'Content-Type: application/json' \
             --data @request.json)

